src/utils.ts

```diff
@@ -1 +1,33 @@
-export const APP_SECRET = '$UP3R S3CR3T';
\ No newline at end of file
+import * as jwt from 'jsonwebtoken';
+import * as HttpStatusCode from 'http-status-codes';
+
+import { AuthPayload } from './resolvers/types';
+import { DetailedError } from './DetailedError';
+
+export const APP_SECRET = '$UP3R S3CR3T';
+
+export const verifyAuthToken = (authorization: string | null, response: { statusCode: number }): void => {
+  if (authorization) {
+    const token = authorization.replace('Bearer ', '');
+    let payload;
+    try {
+      payload = jwt.verify(token, APP_SECRET) as AuthPayload;
+      
+    } catch (error) {
+      response.statusCode = HttpStatusCode.UNAUTHORIZED;
+      throw error;
+    }
+    if (!payload.userId) {
+      response.statusCode = HttpStatusCode.BAD_REQUEST;
```

Comment:
Uma coisa que pode escalar e ficar meio repetitiva é essa questão de setar o statusCode das requests. Para cada erro que você for jogar, você teria que fazer esse passo antes, e em alguns casos inclusive você acabaria setando os mesmos valores pra ele, correto?

Se não estou enganado, o `formatError` recebe esse response como um dos parâmetros, então você conseguiria, baseado no erro que jogou, decidir o código lá dentro do `formatError`, certo?

Reply:
Concordo com a questão de ficar repetitivo, mas não conseguindo imaginar a implementação da proposta:

Segundo a doc do [Apollo Server](https://www.apollographql.com/docs/apollo-server/features/metrics/#high-level-logging) o `formatError` só recebe o `error` como argumento. Existe o `formatResponse` que recebe o `response` como argumento.

O que eu pensei que poderia ser feito seria colocar um campo `status` na mensagem de erro do GraphQL no `formatError` e no `formatResponse` colocar este valor no response de fato.